amazon-s3/6.0/modules/ROOT/pages/amazon-s3-connector-reference.adoc

```diff
@@ -0,0 +1,3652 @@
+
+To ensure that data is not corrupted when traversing the network, specify the Content-MD5 header in the upload part request. Amazon S3 checks the part data against the provided MD5 value. If they do not match, Amazon S3 returns an error.
+
+[NOTE]
+After you initiate a multipart upload and upload one or more parts, you must either complete or abort the multipart upload to stop getting charged for storage of the uploaded parts. After you either complete or abort the multipart upload, Amazon S3 frees up the parts storage and stops charging you for the parts storage.
+
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Bucket name a| String |  Name of the bucket in which the multipart upload is initiated. |  | x
+| Object key a| String |  Object key for which the multipart upload is initiated. |  | x
+| Upload Id a| String |  Upload ID identifying the multipart upload whose part is uploaded. |  | x
+| Part Number a| Number |  Part number of part uploaded. This is a positive integer between 1 and 10,000. |  | x
+| Content a| Any |  Object data. |  | x
+| Content Length a| Number |  Size of the body in bytes. This parameter is useful when the size of the body cannot be determined automatically. |  |
+| Config Ref a| ConfigurationProvider |  Name of the configuration to use to execute this component. |  | x
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  `#[payload]` |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+==== Output
+
+[%autowidth.spread]
+|===
+|Type |<<UploadPartResponse>>
+|===
+
+==== Associated Configurations
+
+* <<config>>
+
+==== Throws
+
+* S3:BAD_REQUEST
+* S3:CONNECTIVITY
+* S3:FORBIDDEN
+* S3:RETRY_EXHAUSTED
+* S3:S3_ERROR
+* S3:SERVER_BUSY
+* S3:SERVICE_ERROR
+
+
+[[UploadPartCopy]]
+=== Upload Part Copy
+`<s3:upload-part-copy>`
+
+
+Uploads a part by copying data from an existing object as data source. The minimum allowable part size for a multipart upload is 5 MB. Instead of using an existing object as part data, you might use the <<UploadPart>> operation and provide data in your request. You must initiate a multipart upload before you can upload any part. Amazon S3 returns an unique identifier, the upload ID, that you must include in your upload part request.
```

Comment:
```suggestion
Uploads a part by copying data from an existing object as data source. The minimum allowable part size for a multipart upload is 5 MB. Instead of using an existing object as part data, you can use the <<UploadPart>> operation and provide data in your request. You must initiate a multipart upload before you can upload a part. Amazon S3 returns a unique identifier, the upload ID, that you must include in your upload part request.
```